lib/internal/bootstrap/pre_execution.js

```diff
@@ -164,6 +166,22 @@ function initializeReportSignalHandlers() {
   addSignalHandler();
 }
 
+function initializeHeapSnapshotSignalHandlers() {
+  const signal = getOptionValue('--heapsnapshot-signal');
+
+  if (!signal)
+    return;
+
+  const { writeHeapSnapshot } = require('v8');
+
+  process.on(signal, () => {
+    if (typeof global.gc === 'function')
```

Comment:
Why is this added? V8 should perform GC twice when a heap snapshot is taken.

Reply:
I was unsure on that bit. Happy to remove it.